tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -2371,6 +2557,437 @@ void test_lte_lc_neighbor_cell_measurement_cancel(void)
 	TEST_ASSERT_EQUAL(-EFAULT, ret);
 }
 
+void test_lte_lc_neighbor_cell_measurement_normal_invalid_field_format_fail(void)
+{
+	int ret;
+	struct lte_lc_ncellmeas_params params = {
+		.search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_LIGHT,
+		.gci_count = 0,
+	};
+
+	/* Syntax:
+	 * %NCELLMEAS: status
+	 * [,<cell_id>,<plmn>,<tac>,<ta>,<earfcn>,<phys_cell_id>,<rsrp>,<rsrq>,<meas_time>
+	 * [,<n_earfcn>1,<n_phys_cell_id>1,<n_rsrp>1,<n_rsrq>1,<time_diff>1]
+	 * [,<n_earfcn>2,<n_phys_cell_id>2,<n_rsrp>2,<n_rsrq>2,<time_diff>2]
+	 * ...
+	 * [,<n_earfcn>17,<n_phys_cell_id>17,<n_rsrp>17,<n_rsrq>17,<time_diff>17]
+	 * [,<timing_advance_measurement_time>]
+	 */
+
+	/* status */
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT%NCELLMEAS", EXIT_SUCCESS);
+	ret = lte_lc_neighbor_cell_measurement(NULL);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	strcpy(at_notif,
+		"%NCELLMEAS:invalid,\"00112233\",\"98712\",\"0AB9\",4800,7,63,31,456,4800,"
+		"8,60,29,4,3500,9,99,18,5,5300\r\n");
+	at_monitor_dispatch(at_notif);
```

Comment:
No event is sent in case of a parsing failure. Generally, parsing should not fail, but we could consider sending the event also in these cases. After all, the API for neighbor cell measurements is sort of a request/response, so the application may be waiting for an event which never comes.

Reply:
That's the way it has been. Generally clients should be prepared that the notification may never come and those few cases in NCS that we can see are fine. They don't get stuck waiting for it forever.